app/helpers/project_helper.rb

```diff
@@ -1,75 +1,78 @@
 # frozen_string_literal: true
 
 module ProjectHelper
-  def event_select_options(event_names, url_params, day_not_allowed)
+  def event_select_options(event_names, url_params, possible_aggregation)
     event_names.map do |e|
       selected = "selected" if e.parameterize == url_params[:event]
-      "<option data-url='#{build_event_url(e, url_params.permit(*Project::PROJECT_PARAMS), day_not_allowed)}' data-name='#{e.parameterize}' #{selected}>#{e}</option>"
+      "<option data-url='#{build_event_url(e, url_params.permit(*Project::PROJECT_PARAMS), possible_aggregation)}' data-name='#{e.parameterize}' #{selected}>#{e}</option>"
     end
   end
 
-  def agg_select_options(url_params, day_not_allowed)
+  def agg_select_options(aggregation, url_params, possible_aggregation)
     Event::AGGREGATIONS.map do |k, v|
-      disabled = "disabled" if k == "d" && day_not_allowed
-      selected = "selected" if k == url_params[:agg] && !disabled
-      "<option data-url='#{build_agg_url(k, url_params.permit(*Project::PROJECT_PARAMS), day_not_allowed)}' data-name='#{k}' #{disabled} #{selected}>#{v.capitalize}</option>"
+      disabled = "disabled" if !possible_aggregation.include?(k) 
+      selected = "selected" if k == aggregation.first && !disabled
```

Comment:
Why do we need `.first` here? And is the `!disabled` check necessary?

Reply:
because aggregation is "month" or "day" but k is "m" or "d", should I do it differently?

Reply:
Ah, out of the box thinking :D But in this case it's not intuitive and rather opaque. It's better to use the hash constant `Event::AGGREGATIONS` to do "conversions".

Reply:
solved it now with:
selected = "selected" if v == aggregation && !disabled

didn't find out how to do it with k == Event::AGGREGATIONS[aggregation] didn't work (but works with key, "w" -> "week", not the other way round)

Reply:
Ah yes, you're right. Can you leave out the `&& disabled`, since if `v == aggregation` it shouldn't be disabled anyways?

Reply:
removed !disabled, not necessary